pkg/routeagent_driver/handlers/calico/ippool_handler.go

```diff
@@ -0,0 +1,256 @@
+/*
+SPDX-License-Identifier: Apache-2.0
+
+Copyright Contributors to the Submariner project.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package calico
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/pkg/errors"
+	calicoapi "github.com/projectcalico/api/pkg/apis/projectcalico/v3"
+	calicocs "github.com/projectcalico/api/pkg/client/clientset_generated/clientset"
+	"github.com/submariner-io/admiral/pkg/log"
+	submV1 "github.com/submariner-io/submariner/pkg/apis/submariner.io/v1"
+	"github.com/submariner-io/submariner/pkg/cni"
+	"github.com/submariner-io/submariner/pkg/event"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/client-go/rest"
+	k8snet "k8s.io/utils/net"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+const (
+	submarinerIPPool = "submariner.io/ippool"
+)
+
+type createUpdateFunc func(ipPoolObj *calicoapi.IPPool) error
+
+type calicoIppoolHandler struct {
+	event.HandlerBase
+	restConfig      *rest.Config
+	cs              *calicocs.Clientset
+	remoteEndpoints map[string]*submV1.Endpoint
+	isGateway       bool
```

Comment:
We should protect these with a mutex as is done in other handlers (eg ).https://github.com/submariner-io/submariner/blob/c2cae00ace9c6c50cb1294a94cd36d75f19f207a/pkg/routeagent_driver/handlers/kubeproxy/gw_transition.go#L29

Reply:
Do you still think we should use mutex here ?

Reply:
As discussed elsewhere it shouldn't be needed unless mutable fields can be accessed outside the parent's event loop. This appears to be the case with `isGateway` in the `Stop` method. To be completely safe, you can use an [atomic bool](https://pkg.go.dev/sync/atomic#Bool) for `isGateway`.